src/Moq/ProxyFactories/IProxy.cs

```diff
@@ -1,15 +1,17 @@
 // Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD.
 // All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
 
-namespace Moq
+using System.ComponentModel;
+
+namespace Moq.Internals
 {
-	/// <summary>
-	///   All proxies created by a <see cref="ProxyFactory"/> should implement this interface.
-	/// </summary>
-	// Interface proxies require this to be able to forward invocations of `object` members
-	// to their `IInterceptor`.
-	internal interface IProxy
+	/// <summary>Do not use. (Moq requires this interface so that <see langword="object"/> methods can be set up on interface mocks.)</summary>
+	// NOTE: This type is actually specific to our DynamicProxy implementation of `ProxyFactory`.
+	// It is directly related to `InterfaceProxy`, see the comment there.
+	[EditorBrowsable(EditorBrowsableState.Never)]
+	public interface IProxy
 	{
-		IInterceptor Interceptor { get; }
+		/// <summary/>
+		object Interceptor { get; }
```

Comment:
Changing this property's type to `object` is unfortunate, but required unless we want to make `IInterceptor` public, too.